pkg/handler/v1/provider/openstack.go

```diff
@@ -28,34 +29,45 @@ import (
 	"k8c.io/kubermatic/v2/pkg/provider"
 )
 
+type credential struct {
+	credential, username, password, tenant, tenantID, domain string
+}
+
+func auth(ctx context.Context, request interface{}, userInfoGetter provider.UserInfoGetter, presetsProvider provider.PresetProvider) (*provider.UserInfo, *credential, error) {
```

Comment:
Any particular reason why you are using interface here?

Reply:
Yes, this method is called by different types of requests e.g., OpenstackReq, OpenstackTenantReq, etc.
So request interface is used to make the method generalized.

Reply:
Would not be simpler to use the minimum common denominator? e.g. OpenstackTenantReq maybe

Reply:
done